explorer-ui/src/components/MatchGrid/AllUsers/AllUsers.jsx

```diff
@@ -0,0 +1,79 @@
+import * as React from "react"
+import "../MatchGrid.css"
+import Geocode from "react-geocode";
+import {
+  Box,
+  Button,
+  ButtonGroup,
+  Flex,
+  HStack,
+  IconButton,
+  Input,
+  SkeletonText,
+  Text,
+} from '@chakra-ui/react'
+Geocode.setApiKey("AIzaSyA4B7q2I3Alla6f8udR0Nr-_3vB8lW5Te0");
+Geocode.setLanguage("en");
+Geocode.setLocationType("ROOFTOP");
+Geocode.enableDebug();
+import MatchCard from "../MatchCard/MatchCard"
+import MapContainer from "../../Mapping/MapContainer";
+import { useState } from "react";
+import { useEffect } from "react";
+import { ChakraProvider, theme } from "@chakra-ui/react";
+export default function AllUsers({profiles}) {
+    const [matchesPlusCoordinates, setMatchesPlusCoordinates] = useState([]);
+    const [userCoordinates, setUserCoordinates] = useState([]);
+    useEffect(() => {
+        getCoordinates(profiles);
+        // getUserCoords(profiles[0])
+    }, [profiles])
+    const getCoordinates = (profiles) => {
+        if (profiles) {
+          for (let i = 0; i < profiles.length; i++) {
+            if(profiles[i].address != null || profiles[i].address!="" ){
+            Geocode.fromAddress(profiles[i].address).then(
+              (response) => {
+                const { lat, lng } = response.results[0].geometry.location;
+                setMatchesPlusCoordinates(matchesPlusCoordinates => [...matchesPlusCoordinates, {id: i, position: {lat, lng}, user: profiles[i]}])
+              },
+              (error) => {
+                console.error(error);
+              }
+            );
+            }
+          }
```

Comment:
nit - promise.all or await the async actions to not have to rely on previousState within the state setter function as harder to reason about the logic